build.go

```diff
@@ -97,7 +97,7 @@ func (c *Client) Build(ctx context.Context, opts BuildOptions) error {
 	lifecycleVersion := descriptor.Info.Version
 	if lifecycleVersion == nil {
 		c.logger.Warnf("lifecycle version unknown, assuming %s", style.Symbol(builder.AssumedLifecycleVersion))
-		lifecycleVersion = builder.VersionMustParse(builder.AssumedLifecycleVersion)
+		// lifecycleVersion = builder.VersionMustParse(builder.AssumedLifecycleVersion)
```

Comment:
This doesn't look right.

Reply:
yeah. Exactly the kind of bugs that a linter catches :-)

Reply:
Sorry, I probably should have been more specific. I was calling out the fact that this line is commented out. What lint warning or error was the linter providing?

Reply:
It says, lifecycleVersion is not used:
```
ineffassign: Detects when assignments to existing variables are not used
```

Reply:
Ah! That makes sense. Probably remnants of prior logic. The message still makes sense but might even need to move to a more appropriate location of the code base.

So this should probably now look like this for the time being:

```go
if descriptor.Info.Version == nil {
	c.logger.Warnf("lifecycle version unknown, assuming %s", style.Symbol(builder.AssumedLifecycleVersion))
} else {
	c.logger.Debugf("Executing lifecycle version %s", style.Symbol(descriptor.Info.Version.String()))
}
```

Agree with your sentiment. Lint would have caught this. 😄